### DISCUSSION TITLE
Prevent delete record in a relation reference not nonempty

### DISCUSSION THREAD

**Discussion by oscar-costa** [pocketbase/pocketbase#1812] [Q&A] [answered]
Hi,
As I understand, it is not possible to delete a record from a collection if this record is referenced in a relational field in another collection. In the case of trying to delete it, an error is displayed: "Failed to delete record. Make sure that the record is not part of a required relation reference."

As I understand, this behavior only applies in case of having selected the option "nonempty" on the relational field options. Is my understanding correct? Is it posible to prevent deleting the record also in case of not having "nonempty" selected?

**Comment by ganigeorgiev** [comment, marked as answer]
You can try with a `@collection.*` "Delete API" rule:

```js
@collection.another.rel != id
```

Note that the above will work only for v0.11+ (_due to the multi-match/match-all support_) and admins will still be able to delete the record (_because API rules don't apply for them_).

Another way could be to manually perform the check by querying the database in a [`OnRecordBeforeDeleteRequest` event hook](https://pocketbase.io/docs/event-hooks/#record-crud-api-hooks) (_or in `OnModelBeforeDelete` hook if you want to handle also any other collections cascade operations_), something like:

```go
app.OnRecordBeforeDeleteRequest("myCollection").Add(func(e *core.RecordDeleteEvent) error {
	// check for at least one ref collection record pointing to the currently deleted record
	refRecord, err := app.Dao().FindFirstRecordByData("refCollection", "relField", e.Record.Id)
	if err != nil || refRecord != nil {
		return apis.NewBadRequestError("The record is part of a relation and cannot be deleted", err)
	}

	return nil
})
```

**Reply by oscar-costa** [reply]
the "@collection.another.rel != id" approach worked for me

Thanks
